scipy/sparse/dok.py

```diff
@@ -492,11 +492,7 @@ def tocsc(self, copy=False):
     tocsc.__doc__ = spmatrix.tocsc.__doc__
 
     def resize(self, shape):
-        """Resize the matrix in-place to dimensions given by `shape`.
-
-        Any non-zero elements that lie outside the new shape are removed.
-        """
-        if not isshape(shape):
+        if not isshape(shape, nonneg=2):
             raise TypeError("Dimensions must be a 2-tuple of positive integers")
```

Comment:
I know the existing DOK implementation raised a `TypeError`, but this case seems more like a `ValueError` to me.